File: android/src/main/java/ch/nevis/mobile/authentication/sdk/react/model/message/outgoing/LocalAuthenticatorsOutMessage.kt
/*
 * Copyright © 2023 Nevis Security AG. All rights reserved.
 */

package ch.nevis.mobile.authentication.sdk.react.model.message.outgoing

import ch.nevis.mobile.authentication.sdk.react.model.sdk.AuthenticatorImpl
import kotlinx.serialization.Serializable

@Serializable
data class LocalAuthenticatorsOutMessage(
    override val operationId: String,
    val authenticators: List<AuthenticatorImpl>
) : ChannelOutMessage()

File: ios/Model/Message/Outgoing/LocalAuthenticatorsOutMessage.swift
//
// Copyright © 2023 Nevis Security AG. All rights reserved.
//

import NevisMobileAuthentication

class LocalAuthenticatorsOutMessage: ChannelOutMessage {
	// MARK: Properties

	let authenticators: [TypedAuthenticator]

	// MARK: Initialization

	init(operationId: String, authenticators: [any Authenticator]) {
		self.authenticators = authenticators.map { TypedAuthenticator(wrapped: $0) }
		super.init(operationId: operationId)
	}

	// MARK: Encodable

	enum CodingKeys: String, CodingKey {
		case authenticators
	}
	
	override func encode(to encoder: Encoder) throws {
		var container = encoder.container(keyedBy: CodingKeys.self)
		try container.encode(authenticators, forKey: .authenticators)
		try super.encode(to: encoder)
	}
}
